sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/CombiningLimitsSuite.scala

```diff
@@ -117,4 +128,111 @@ class CombiningLimitsSuite extends PlanTest {
       testRelation.select().groupBy()(count(1)).orderBy(count(1).asc).analyze)
     comparePlans(optimized4, expected4)
   }
+
+  test("SPARK-33497: Eliminate Limit if LocalRelation max rows not larger than Limit") {
+    checkPlan(
+      testRelation.select().limit(10),
+      testRelation.select(),
+      10
+    )
+  }
+
+  test("SPARK-33497: Eliminate Limit if Range max rows not larger than Limit") {
+    checkPlan(
+      Range(0, 100, 1, None).select().limit(200),
+      Range(0, 100, 1, None).select(),
+      100
+    )
+    checkPlan(
+      Range(-1, Long.MaxValue, 1, None).select().limit(1),
+      Range(-1, Long.MaxValue, 1, None).select().limit(1),
+      1
+    )
+  }
+
+  test("SPARK-33497: Eliminate Limit if Sample max rows not larger than Limit") {
+    checkPlan(
+      testRelation.select().sample(upperBound = 0.2, seed = 1).limit(10),
+      testRelation.select().sample(upperBound = 0.2, seed = 1),
+      10
+    )
+  }
+
+  test("SPARK-33497: Eliminate Limit if Deduplicate max rows not larger than Limit") {
+    checkPlan(
+      testRelation.deduplicate(Symbol("a")).limit(10),
+      testRelation.deduplicate(Symbol("a")),
+      10
+    )
+  }
+
+  test("SPARK-33497: Eliminate Limit if Repartition max rows not larger than Limit") {
+    checkPlan(
+      testRelation.repartition(2).limit(10),
+      testRelation.repartition(2),
+      10
+    )
+    checkPlan(
+      testRelation.distribute(Symbol("a"))(2).limit(10),
+      testRelation.distribute(Symbol("a"))(2),
+      10
+    )
+  }
+
+  test("SPARK-33497: Eliminate Limit if Join max rows not larger than Limit") {
```

Comment:
It seems this test unit has similar tests, so could you make it more compact just like this?
```
Seq(Inner, FullOuter, LeftOuter, ...).foreach { joinType =>
  checkPlan(...)
}
```

Reply:
yeah it looks better, thanks